tests/modify.spec.ts

```diff
@@ -0,0 +1,182 @@
+      types: [
+        interfaceType({
+          name: 'Node',
+          definition(t) {
+            t.id('id', {
+              description: 'Some Node ID Description',
+              resolve: () => {
+                throw new Error('Abstract')
+              },
+            })
+            t.field('subNode', {
+              type: 'Node',
+            })
+          },
+          resolveType: (o) => o.__typename,
+        }),
+        objectType({
+          name: 'User',
+          definition(t) {
+            t.implements('Node')
+            t.modify('id', {
+              description: 'Some User ID Description',
+              resolve: (root) => `User:${root.id}`,
+            })
+            t.field('subNode', {
+              type: 'User',
+            })
+          },
+        }),
+        objectType({
+          name: 'Throws',
+          definition(t) {
+            t.implements('Node')
+          },
+        }),
+        objectType({
+          name: 'AddArg',
+          definition(t) {
+            t.implements('Node')
+            t.modify('id', {
+              args: { someArg: stringArg() },
+              resolve: (root, args) => args.someArg ?? `AddArg:${root.id}`,
+            })
+          },
+        }),
+        queryField('node', {
+          type: 'Node',
+          args: {
+            id: idArg({ required: true }),
+          },
+          resolve: (root, args) => {
+            const [__typename, id] = args.id.split(':')
+            return { __typename, id }
+          },
+        }),
+      ],
+      outputs: false,
+      shouldGenerateArtifacts: false,
+      features: {
+        abstractTypeStrategies: {
+          resolveType: true,
+        },
+      },
+    })
+  })
+
+  it('can modify an existing interface field description', async () => {
+    const result = await graphql(
+      schema,
+      `
+        {
+          node: __type(name: "Node") {
+            fields {
+              name
+              description
+            }
+          }
+          user: __type(name: "User") {
+            fields {
+              name
+              description
+            }
+          }
+        }
+      `
+    )
+    expect(result.data?.node.fields.find((f) => f.name === 'id').description).toEqual(
+      'Some Node ID Description'
+    )
+    expect(result.data?.user.fields.find((f) => f.name === 'id').description).toEqual(
+      'Some User ID Description'
+    )
+  })
+
+  it('can modify an existing interface field resolve', async () => {
+    const result = await graphql(
+      schema,
+      `
+        {
+          user: node(id: "User:1") {
+            id
+          }
+          throws: node(id: "Throws:1") {
+            id
+          }
+        }
+      `
+    )
+    expect(result.data?.user.id).toEqual('User:1')
+    expect(result.data?.throws.id).toEqual(null)
+    expect(result.errors?.[0].message).toEqual('Abstract')
+  })
+
+  it('can add additional field args to an interface field', async () => {
```

Comment:
Do we have a test to show that it can modify an existing field arg to an interface field? A modification for example being a change of arg type or arg description.

Reply:
See other comment, no its not supported since it would violate the interface implementation.

I don't think we can enforce this statically though can we? E.g. blacklist fields. Instead we'll have to do runtime checks won't we?